public/content/posts/performance-improvement-in-a-next.js-page/index.md

```diff
@@ -0,0 +1,120 @@
+
+### improve performance of the image
+
+Once we identify the LCP element of the page, we can try to make sure it loads fast relatively. Since in this case it's an image we can should load the image in "eager" mode instead of the "lazy" mode.
+
+```html
+<img src="imageUrl" loading="eager" />
+```
+
+since the eager is the default value for loading prop, in this case we can ignore the loading prop as well.
+
+[This Article](/posts/improving-images-on-fishbrain.com) from our blog also explains how we have improved the overall performance of the images.
+
+### Preload the LCP image
+
+Perloading the image inside next/head is another effective way to improve the LCP score.
+
+```jsx
+<Head>
+  <link rel="preload" href={imageUrl} as="image" />
+</Head>
+```
+
+This will ask the browser to load image upfront even before it discovers the image element.
+
+For testing a single page (url) performance, [web.dev/measure](https://web.dev/measure) is a great tool. After measuring the performance with this tool, it provides us some useful methods on how to improve our web vitals scores.
+
+### Next.js Dynamic imports
+
+One of the method it suggest is to reduce the initial response time. For this we could use [Next.js dynamic imports](https://nextjs.org/docs/advanced-features/dynamic-import). It's a handy way to load components and libraries only when those are needed. Therefore we could use the dynamic imports when we are loading components conditionaly. This will also slightly reduce the build size.
+
+```jsx
+const DynamicCatchesSection = dynamic(async () =>
+  import("./components/CatchesSection")
+);
+```
+
+```jsx
+<DynamicCatchesSection
+  title="This is title"
+  regionCode={regionCode}
+  slug={slug}
+  countryCode={countryCode}
+/>
+```
+
+### Results
+
+Before:
+![Lighthouse Performance Desktop Before](/performance.png "Lighthouse Performance Desktop Before")
+
+After:
+![Lighthouse Performance Desktop After](/performance-2.png "Lighthouse Performance Desktop After")
+As you can see performance is quite good and specially the LCP score has improved a lot compared to the earlier version.
+
+**Note:** _The performance is tested in lighthouse tool for Desktop type. Score can be different from time to time and for mobile devices this score will be considerably lower since the mobile has less resource power than the Desktop devices._
+
+Below are some few things you can try to improve the overall performance of a Next.js application.
+
+### Next.js image
+
+Next.js also has provide us a good image component (next/image). [Check it out here](https://nextjs.org/docs/api-reference/next/image) And they are also experimenting on a new image component - [next/future/image](https://nextjs.org/docs/api-reference/next/future/image) with the help from user's feedback. It will be good to keep an eye on this one since it might help us to improve the performance of the image even more.
```

Comment:
```suggestion
Next.js also has provide us a good image component (next/image). [Check it out here](https://nextjs.org/docs/api-reference/next/image) They are also experimenting on a new image component - [next/future/image](https://nextjs.org/docs/api-reference/next/future/image) with the help from user's feedback. It will be good to keep an eye on this one since it might help us to improve the performance of the image even more.
```